core/unit_test/TestScan.hpp

```diff
@@ -101,7 +101,7 @@ struct TestScan {
     int64_t total = 0;
     Kokkos::parallel_scan(N, *this, total);
 
-    run_check(size_t((N + 1) * N / 2), size_t(total));
+    [&] { ASSERT_EQ(size_t((N + 1) * N / 2), size_t(total)); }();
```

Comment:
what is this? if you get rid of the run_check (which I don't object to at all) why exactly do you create a lambda?

Reply:
Essentially, this is to ignore `ASSERT_EQ`'s return value. Otherwise, the compiler is complaining that constructors can't return a value.

Reply:
is that better than the run_check function?

Reply:
A little bit? I don't feel strongly about it and can revert if you prefer.

Reply:
Revert or add a comment